### ISSUE TITLE
Workflow new events

### ISSUE THREAD

**Issue by danieltudosiu** [Project-MONAI/MONAI#1917]
**Is your feature request related to a problem? Please describe.**
Pre-loss.backward event.

**Describe the solution you'd like**
Make the Engines fire an event between the loss and backward pass so that handlers that manage gradients can be triggered more elegantly. 

**Describe alternatives you've considered**
Allow the loss function to trigger events by passing the fire_event callable. It breaks the Ignite paradigm since the loss function will be aware of the Engine.

**Additional context**
This will be required for work such as GradNorm and adaptively weighting Discriminator's gradients for the Generator in GANs.

**Comment by wyli** [comment]
Hi @danieltudosiu there're some additional events: https://github.com/Project-MONAI/MONAI/blob/32e540f99d43d56c038537dd2ed6954d80a531ee/monai/engines/utils.py#L33-L40
would `LOSS_COMPLETED` already address this issue?

**Comment by danieltudosiu** [comment]
Hi @wyli ,

Yes, loss completed would solve this.

Are they available in 0.4?


Thanks!

**Comment by wyli** [comment]
no, this will be available in v0.5 which should be released in the next few weeks :) I'm closing this ticket
